pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -51,49 +53,86 @@ type MatrixInclude struct {
 	Params Params `json:"params,omitempty"`
 }
 
-// FanOut produces combinations of Parameters of type String from a slice of Parameters of type Array.
-func (m *Matrix) FanOut() []Params {
-	var combinations []Params
-	for _, parameter := range m.Params {
-		combinations = fanOut(parameter, combinations)
+// Combination holds a single combination from a Matrix with key as param.Name and value as param.Value
+type Combination map[string]string
+
+// Combinations holds a list of combination for a given Matrix
```

Comment:
```suggestion
// Combinations is a slice of Combinations
```

Suggesting this because combinations doesn't reference the matrix that created them